rsc/src/main/scala/rsc/Compiler.scala

```diff
@@ -21,14 +21,23 @@ import rsc.syntax._
 import rsc.util._
 
 class Compiler(val settings: Settings, val reporter: Reporter) extends AutoCloseable with Pretty {
+
+  private val startInit = System.nanoTime()
+
   var trees: List[Source] = Nil
   var gensyms: Gensyms = Gensyms()
-  var classpath: Classpath = Classpath(settings.cp)
+  var classpath: Classpath = profile(settings, reporter, "classpath") { Classpath(settings.cp) }
   var symtab: Symtab = Symtab(classpath)
   var todo: Todo = Todo()
   var infos: Infos = Infos(classpath)
   var output: Output = Output(settings)
 
+  private val endInit = System.nanoTime()
+  private val msInit = (endInit - startInit) / 1000000
+  if (settings.xprint("timings")) {
```

Comment:
It may be cleaner to setup the reporter with the settings, then you won't have to keep checking your xprint settings, you would basically have a NullReporter if the settings were disabled.